ArtOfIllusion/src/artofillusion/animation/JointEditorDialog.java

```diff
@@ -348,72 +401,166 @@ public DOFGraph getGraph()
   private class DOFGraph extends CustomWidget
   {
     private DOFPanel panel;
-    private int dragging;
+    private boolean dragging0, dragging1, dragging2, dragging3, dragging4;
```

Comment:
Any particular reason that you broke these up into multiple Booleans? If I understand correctly, only one of them can be true at a time. If that supposition is true, a more idiomatic structure here would be a `private enum` of the various dragging 'modes'. (With descriptive names)

Reply:
I guess the core question is: How many of these booleans can be true at the same time?

Reply:
Actually, when I built it the way I did I had a slightly different plan on how to decide which handle to move. Also the graphic appearance is pretty much parametrized. It was possible to have the handles overlap in their initial positions more than they do now.

I can change it back closer to what it was.

Reply:
These are each one of the handles, correct? So Joint, CW_hardstop, CCW_hardstop, CW_softstop & CCW_softstop?

If you're moving one of the hard-stops, the update code should, of course, validate if the  soft-stops and the joint are still at valid locations. You are still only *dragging* one handle.

This is, in fact, almost a perfect textbook case of what the `enum` structure was invented for. I think @peastman used an `int` originally because `enum` was not part of the java language yet. (That's an old, C-lang way of faking an enum)

Would you like me to run up a PR against your branch to show you what I mean?

At the very least, we should give each of them a descriptive name...

Reply:
> You are still only dragging one handle.
> give each of them a descriptive name...

Yes, I was just deriving from the existing code but logically it should separate `onHandle` and `dragging` tell better exactly what.

> Would you like me to run up a PR against your branch to show you what I mean?

Go ahead. :)

Reply:
I'm actually going to leave this sub-project for now. I still think that something along those lines is a better target, but I don't want to hold up these user interface changes anymore.

I'll come back to it once we've got a reasonable release finished.